File: poc-kronos-frontend/src/model/pessoa/EnderecoPessoa.js
class EnderecoPessoa {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.cep = dados ? dados.cep : null;
    this.logradouro = dados ? dados.logradouro : null;
    this.complemento = dados ? dados.complemento : null;
    this.bairro = dados ? dados.bairro : null;
    this.localidade = dados ? dados.localidade : null;
    this.uf = dados ? dados.uf : null;
    this.unidade = dados ? dados.unidade : null;
    this.ibge = dados ? dados.ibge : null;
    this.gia = dados ? dados.gia : null;
    this.idPessoa = dados ? dados.idPessoa : null;
    this.idTipoEndereco = dados ? dados.idTipoEndereco : null;
    this.nomeTipoEndereco = dados ? dados.nomeTipoEndereco : null;
    this.numero = dados ? dados.numero : null;
  }

  adicionarDadosCep(dados) {
    this.cep = dados.cep.replace("-", "");
    this.logradouro = dados.logradouro;
    this.complemento = dados.complemento;
    this.bairro = dados.bairro;
    this.localidade = dados.localidade;
    this.uf = dados.uf;
    this.unidade = dados.unidade;
    this.ibge = dados.ibge;
    this.gia = dados.gia;
  }

  static validations(required, maxLength, minLength, numeric) {
    return {
      cep: {
        required,
        maxLength: maxLength(8),
        minLength: minLength(8),
        numeric
      },
      logradouro: { required },
      bairro: { required },
      localidade: { required },
      uf: { required },
      ibge: { required },
      idPessoa: { required },
      idTipoEndereco: { required }
    };
  }

  static validationMessages() {
    return {
      cep: {
        required: "CEP é obrigatório",
        maxLength: "CEP deve ter 8 digítos",
        minLength: "CEP deve ter 8 digítos",
        numeric: "CEP só aceita digítos"
      },
      logradouro: {
        required: "Logradouro é obrigatório"
      },
      bairro: {
        required: "Bairro é obrigatório"
      },
      localidade: {
        required: "Localidade é obrigatória"
      },
      uf: {
        required: "UF é obrigatória"
      },
      ibge: {
        required: "IBGE é obrigatório"
      },
      idPessoa: {
        required: "Pessoa é obrigatória"
      },
      idTipoEndereco: {
        required: "Tipo Endereço é obrigatório"
      }
    };
  }
}
export default EnderecoPessoa;

File: src/main/java/br/com/kronos/backend/aplicacao/pessoa/EnderecoPessoa.java
package br.com.kronos.backend.aplicacao.pessoa;

import static java.util.Objects.requireNonNull;

import lombok.Builder;
import lombok.Getter;

@Getter
public class EnderecoPessoa {
	
	private int id;
	private String cep;	
	private String logradouro;
    private String complemento;
    private String bairro;
	private String localidade;
    private String uf;
    private String unidade;
    private String ibge;
    private String gia;
    private int idPessoa;
	private int idTipoEndereco;
    private String numero;
    	
	@Builder
    public EnderecoPessoa(int id, String cep, String logradouro, String complemento, String bairro, String localidade, String uf, 
                           String unidade, String ibge, String gia, int idPessoa, int idTipoEndereco, String numero) {
        super();
        this.id = id;
        this.cep = requireNonNull(cep, "cep é obrigatório");		
        this.logradouro = requireNonNull(logradouro, "logradouro é obrigatório");	
        this.complemento = complemento;
        this.bairro = requireNonNull(bairro, "bairro é obrigatório");	
        this.localidade = requireNonNull(localidade, "localidade é obrigatório");	
        this.uf = requireNonNull(uf, "uf é obrigatório");
        this.unidade = unidade;
        this.ibge = requireNonNull(ibge, "ibge é obrigatório");	
        this.gia = gia;
        this.idPessoa = requireNonNull(idPessoa, "idPessoa é obrigatório");	
        this.idTipoEndereco = requireNonNull(idTipoEndereco, "idTipoEndereco é obrigatório");	
        this.numero = numero;

	}

}